lib/process-release.js

```diff
@@ -0,0 +1,87 @@
+var semver = require('semver')
+var url = require('url')
+var path = require('path')
+
+var bitsre = /(x32|x64)/
+
+function processRelease (argv, gyp, defaultVersion, defaultRelease) {
+  var versionStr = argv[0] || gyp.opts.target || defaultVersion
+
+  // parse the version to normalize and ensure it's valid
+  var versionSemver = semver.parse(versionStr)
+  if (!versionSemver) {
+    return { versionStr: versionStr }
+  }
+  // flatten version into String
+  versionStr = versionSemver.version
+
+  var overrideDistUrl = gyp.opts['dist-url'] || gyp.opts.disturl
+  var iojs = !overrideDistUrl && !defaultRelease && semver.gte(versionStr, '1.0.0')
+  var name = (defaultRelease && defaultRelease.name) || (iojs ? 'iojs' : 'node')
+  var defaultDirUrl = (overrideDistUrl || iojs ? 'https://iojs.org/download/release' : 'https://nodejs.org/dist') + '/v' + versionStr + '/'
```

Comment:
Seems like this should go below the `process.release` branch, if it wasn't for [lines 28 & 29](https://github.com/nodejs/node-gyp/pull/711/files#r38764357)